data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/contingency/CsaProfileContingencyCreationContext.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package com.farao_community.farao.data.crac_creation.creator.csa_profile.crac_creator.contingency;
+
+import com.farao_community.farao.data.crac_creation.creator.api.ElementaryCreationContext;
+import com.farao_community.farao.data.crac_creation.creator.api.ImportStatus;
+
+/**
+ * @author Jean-Pierre Arnould {@literal <jean-pierre.arnould at rte-france.com>}
+ */
+public final class CsaProfileContingencyCreationContext implements ElementaryCreationContext {
+    private String nativeId;
+    private String contingencyId;
+    private String contingencyName;
+    private ImportStatus importStatus;
+    private String importStatusDetail;
+    private boolean isAltered;
+
+    private CsaProfileContingencyCreationContext(String nativeId, String contingencyId, String contingencyName, ImportStatus importStatus, String importStatusDetail, boolean isAltered) {
+        this.nativeId = nativeId;
+        this.contingencyId = contingencyId;
+        this.contingencyName = contingencyName;
+        this.importStatus = importStatus;
+        this.importStatusDetail = importStatusDetail;
+        this.isAltered = isAltered;
+    }
+
+    public static CsaProfileContingencyCreationContext imported(String nativeId, String contingencyId, String contingencyName, String importStatusDetail, boolean isAltered) {
+        return new CsaProfileContingencyCreationContext(nativeId, contingencyId, contingencyName, ImportStatus.IMPORTED, importStatusDetail, isAltered);
+    }
+
+    public static CsaProfileContingencyCreationContext notImported(String nativeId, String contingencyId, String contingencyName, ImportStatus importStatus, String importStatusDetail) {
+        return new CsaProfileContingencyCreationContext(nativeId, contingencyId, contingencyName, importStatus, importStatusDetail, false);
+    }
```

Comment:
When the contingency is not imported, you better set contingencyId (and contingencyName) to null. So you can remove them from the arguments of this static method and set them always to null.

Reply:
Ok